spectral_cube/wcs_utils.py

```diff
@@ -481,3 +481,189 @@ def diagonal_wcs_to_cdelt(mywcs):
         del mywcs.wcs.cd
         mywcs.wcs.cdelt = cdelt
     return mywcs
+
+
+def find_spatial_pixel_index(cube, xlo, xhi, ylo, yhi):
+    '''
+    Given low and high cuts, return the pixel coordinates for a rectangular
+    region in the given cube or spatial projection. lo and hi inputs can be
+    given in pixels, "min"/"max", or in world coordinates.
+
+    When spatial WCS dimensions are given as an `~astropy.units.Quantity`,
+    the spatial coordinates of the 'lo' and 'hi' corners are solved together.
+    This minimizes WCS variations due to the sky curvature when slicing from
+    a large (>1 deg) image.
+
+
+    Parameters
+    ----------
+    cube : :class:`~SpectralCube` or spatial :clas:`~Projection`
+        A spectral-cube or projection/slice with spatial dimensions.
+    [xy]lo/[xy]hi : int or :class:`~astropy.units.Quantity` or ``min``/``max``
+        The endpoints to extract.  If given as a quantity, will be
+        interpreted as World coordinates.  If given as a string or
+        int, will be interpreted as pixel coordinates.
+
+    Returns
+    -------
+    limit_dict : dict
+        Pixel coordinates of [xy]lo/[xy]hi in the given `cube`.
```

Comment:
```suggestion
        Pixel coordinates of [xy]lo/[xy]hi in the given ``cube``.
```
presumably `cube` isn't a valid rst target, we need double-backticks?